### ISSUE TITLE
make doc does not work with (BSD grep) 2.5.1-FreeBSD in MacOS 10.8.5

### ISSUE THREAD

**Issue by skyl** [jstat/jstat#126]
On MacOS 10.8.5, the builtin version of `grep` is `grep (BSD grep) 2.5.1-FreeBSD`. This version appears not to support the `-P` flag. Thus, `make doc` fails because of:

```
JS_ENGINE ?= $(shell which node nodejs 2>/dev/null | grep -Po -m 1 "(.+?)$$")
```

with:

```
± make doc
usage: grep [-abcDEFGHhIiJLlmnOoPqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
Generating documentation
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
/bin/sh: build/doctool.js: Permission denied
make: *** [doc] Error 126
```

This can be worked around by installing GNU grep and aliasing `grep` to that. For instance, http://stackoverflow.com/a/26064330/177293

**Comment by jamescgibson** [comment]
My bad. I'll figure out how to accomplish the same thing without the -P flag. Thanks for the bug report!

**Comment by trevnorris** [comment]
Fixed by f84ed22. Published in 1.1.3.
